backend/src/main/java/com/redhat/cloud/notifications/routers/EventResource.java

```diff
@@ -130,4 +133,48 @@ public Page<EventLogEntry> getEvents(@Context SecurityContext securityContext, @
         page.setLinks(links);
         return page;
     }
+
+    static EventLogEntryActionStatus fromNotificationStatus(NotificationStatus status) {
+        switch (status) {
+            case SENT:
+                return EventLogEntryActionStatus.SENT;
+            case SUCCESS:
+                return EventLogEntryActionStatus.SUCCESS;
+            case PROCESSING:
+                return EventLogEntryActionStatus.PROCESSING;
+            case FAILED_EXTERNAL:
+            case FAILED_INTERNAL:
+                return EventLogEntryActionStatus.FAILED;
+            default:
+                Log.warnf("Uncovered status found:[%s]. This is a bug", status);
+                return EventLogEntryActionStatus.UNKNOWN;
+        }
+    }
+
+    static Set<NotificationStatus> toNotificationStatus(Set<EventLogEntryActionStatus> statusSet) {
+        Set<NotificationStatus> notificationStatusSet = new HashSet<>();
+        statusSet.forEach(status -> {
+            switch (status) {
```

Comment:
Could you please add a null-check on `status` here? RESTEasy allows `null` values in collections that are received as REST params. We'll probably need to throw a `BadRequestException` is status is `null`.

Reply:
done